plugins/InspectionGadgets/src/com/siyeh/ig/redundancy/RedundantStringOperationInspection.java

```diff
@@ -159,6 +165,80 @@ private ProblemDescriptor getStringConstructorProblem(PsiNewExpression expressio
+
+      PsiExpression equalTo = call.getArgumentList().getExpressions()[0];
+
+      //case: "foo".equals(s.toLowerCase())
+      if (equalTo instanceof PsiMethodCallExpression) {
+        PsiMethodCallExpression equalsToCallExpression = (PsiMethodCallExpression)equalTo;
+        if (isChangeCaseCall(equalsToCallExpression) &&
+            ExpressionUtils.computeConstantExpression(call.getMethodExpression().getQualifierExpression()) instanceof String) {
+          PsiElement anchor = equalsToCallExpression.getMethodExpression().getReferenceNameElement();
+          if (anchor == null) {
+            return null;
+          }
+          return createProblem(equalsToCallExpression, anchor, RemoveRedundantChangeCaseFix.PlaceCaseEqualType.RIGHT);
+        }
+      }
+
+      PsiMethodCallExpression qualifierCall = MethodCallUtils.getQualifierMethodCall(call);
+      if (qualifierCall == null) return null;
+      PsiExpression receiver = qualifierCall.getMethodExpression().getQualifierExpression();
+      if (receiver == null) return null;
+
+      //cases:
+      //- text1.toLowerCase().equals("test2")
+      //- text1.toLowerCase().equals(text2.toLowerCase())
+      if (
+        isChangeCaseCall(qualifierCall)) {
+        PsiElement anchor = qualifierCall.getMethodExpression().getReferenceNameElement();
+        if (anchor == null) {
+          return null;
+        }
+        //case: text1.toLowerCase().equals("test2")
+        if (ExpressionUtils.computeConstantExpression(equalTo) instanceof String) {
+          return createProblem(qualifierCall, anchor, RemoveRedundantChangeCaseFix.PlaceCaseEqualType.LEFT);
+        }
+
+        //case: text1.toLowerCase().equals(text2.toLowerCase())
+        if (equalTo instanceof PsiMethodCallExpression) {
+          PsiMethodCallExpression secondCall = (PsiMethodCallExpression)equalTo;
+          if (isEqualChangeCaseCall(qualifierCall, secondCall)) {
+            return createProblem(secondCall, anchor, RemoveRedundantChangeCaseFix.PlaceCaseEqualType.BOTH);
+          }
+        }
+      }
+
+      return null;
+    }
+
+    @NotNull
+    private ProblemDescriptor createProblem(PsiMethodCallExpression equalsToCallExpression,
+                                            PsiElement anchor,
+                                            RemoveRedundantChangeCaseFix.PlaceCaseEqualType type) {
+      return myManager.createProblemDescriptor(anchor, (TextRange)null,
+                                               InspectionGadgetsBundle.message("inspection.redundant.string.call.may.be.message"),
+                                               ProblemHighlightType.LIKE_UNUSED_SYMBOL, myIsOnTheFly,
+                                               new RemoveRedundantChangeCaseFix(getChangeCaseType(equalsToCallExpression), type));
+    }
+
+
+    private static boolean isChangeCaseCall(@NotNull PsiMethodCallExpression qualifierCall) {
```

Comment:
I prefer creating a composite CallMatcher using anyOf. In this case, the method will be like `return CHANGE_CASE.test(qualifierCall)`, so it could be inlined.

Reply:
Changed with anyOf